From d9b03f1d2f4205425cdfc3a6be5476fc9992de0d Mon Sep 17 00:00:00 2001
From: Adrian Alonso <aalonso@freescale.com>
Date: Thu, 4 Oct 2012 10:38:48 -0500
Subject: ENGR00227020: devices-common imx_dma_data duplicate struct
 definition

* Fix imx_dma_data duplicate struct definition
* Rename struct as name conflicts with imx_dma_data
  struct defined at arch/arm/plat-mxc/include/mach/dma.h
* Update copyrigth year.

Signed-off-by: Adrian Alonso <aalonso@freescale.com>
---
 arch/arm/mach-mx5/devices-imx50.h               |    6 +++---
 arch/arm/mach-mx6/devices-imx6q.h               |    4 ++--
 arch/arm/plat-mxc/devices/platform-dma.c        |    8 ++++----
 arch/arm/plat-mxc/include/mach/devices-common.h |    4 ++--
 4 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-mx5/devices-imx50.h b/arch/arm/mach-mx5/devices-imx50.h
index bdaf322..8cba79c 100755
--- a/arch/arm/mach-mx5/devices-imx50.h
+++ b/arch/arm/mach-mx5/devices-imx50.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2010-2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2010-2012 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -29,8 +29,8 @@ extern const struct imx_srtc_data imx50_imx_srtc_data __initconst;
 #define imx50_add_srtc()	\
 	imx_add_srtc(&imx50_imx_srtc_data)
 
-extern const struct imx_dma_data imx50_dma_data __initconst;
-#define imx50_add_dma()	imx_add_dma(&imx50_dma_data);
+extern const struct imx_dma_res_data imx50_dma_res_data __initconst;
+#define imx50_add_dma()	imx_add_dma(&imx50_dma_res_data);
 
 extern const struct imx_fec_data imx50_fec_data;
 #define imx50_add_fec(pdata)	\
diff --git a/arch/arm/mach-mx6/devices-imx6q.h b/arch/arm/mach-mx6/devices-imx6q.h
index 73f7f7a..8595f7f 100644
--- a/arch/arm/mach-mx6/devices-imx6q.h
+++ b/arch/arm/mach-mx6/devices-imx6q.h
@@ -36,8 +36,8 @@ imx6q_anatop_thermal_imx_data __initconst;
 #define imx6q_add_anatop_thermal_imx(id, pdata)	\
 	imx_add_anatop_thermal_imx(&imx6q_anatop_thermal_imx_data, pdata)
 
-extern const struct imx_dma_data imx6q_dma_data __initconst;
-#define imx6q_add_dma()	imx_add_dma(&imx6q_dma_data);
+extern const struct imx_dma_res_data imx6q_dma_res_data __initconst;
+#define imx6q_add_dma()	imx_add_dma(&imx6q_dma_res_data);
 
 #define imx6q_add_gpmi(platform_data)	imx_add_gpmi(platform_data);
 
diff --git a/arch/arm/plat-mxc/devices/platform-dma.c b/arch/arm/plat-mxc/devices/platform-dma.c
index b2cba3d..7f73195 100644
--- a/arch/arm/plat-mxc/devices/platform-dma.c
+++ b/arch/arm/plat-mxc/devices/platform-dma.c
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2011 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2011-2012 Freescale Semiconductor, Inc. All Rights Reserved.
  *
  * This program is free software; you can redistribute it and/or modify it under
  * the terms of the GNU General Public License version 2 as published by the
@@ -13,19 +13,19 @@
 #include <mach/devices-common.h>
 
 #ifdef CONFIG_SOC_IMX50
-const struct imx_dma_data imx50_dma_data __initconst = {
+const struct imx_dma_res_data imx50_dma_res_data __initconst = {
 	.iobase = MX50_APBHDMA_BASE_ADDR,
 };
 #endif
 
 #ifdef CONFIG_SOC_IMX6Q
-const struct imx_dma_data imx6q_dma_data __initconst = {
+const struct imx_dma_res_data imx6q_dma_res_data __initconst = {
 	.iobase = APBH_DMA_ARB_BASE_ADDR,
 };
 #endif
 
 struct platform_device *__init imx_add_dma(
-			const struct imx_dma_data *data)
+			const struct imx_dma_res_data *data)
 {
 	struct resource res[] = {
 		{
diff --git a/arch/arm/plat-mxc/include/mach/devices-common.h b/arch/arm/plat-mxc/include/mach/devices-common.h
index ae7f8cb..1ff68bd 100755
--- a/arch/arm/plat-mxc/include/mach/devices-common.h
+++ b/arch/arm/plat-mxc/include/mach/devices-common.h
@@ -25,10 +25,10 @@ static inline struct platform_device *imx_add_platform_device(
 			name, id, res, num_resources, data, size_data, 0);
 }
 
-struct imx_dma_data {
+struct imx_dma_res_data {
 	resource_size_t iobase;
 };
-struct platform_device *__init imx_add_dma(const struct imx_dma_data *data);
+struct platform_device *__init imx_add_dma(const struct imx_dma_res_data *data);
 
 #include <linux/fec.h>
 struct imx_fec_data {
-- 
1.7.9.5

